server.go

```diff
@@ -591,7 +591,18 @@ func (s *Server) HandleTracePacket(packet []byte) {
 		log.WithError(err).Warn("Trace unmarshaling error")
 		return
 	}
-
+	for _, metricPacket := range newSample.Metrics {
+		metric, err := samplers.ParseMetricSSF(metricPacket)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				logrus.ErrorKey: err,
+				"packet":        metricPacket.String(),
+			}).Warn("Could not parse packet")
+			s.Statsd.Count("packet.error_total", 1, []string{"packet_type:ssf_metric", "reason:parse"}, 1.0)
+			return
+		}
+		s.Workers[metric.Digest%uint32(len(s.Workers))].PacketChan <- *metric
```

Comment:
The channel is currently unbuffered and we don't expect to see any more metrics coming in this way than per-worker, so I think it's safe to leave it as is for now because it doesn't make anything worse.

But you do raise a good point: These channels can currently back up if single worker is "hot" and we have no real way of knowing this. We should likely come up with a way of measuring this and then we could properly make things buffered?